engine/execution/computation/query/executor.go

```diff
@@ -128,67 +127,65 @@ func (e *QueryExecutor) ExecuteScript(
 	requestCtx, cancel := context.WithTimeout(ctx, e.config.ExecutionTimeLimit)
 	defer cancel()
 
-	scriptInContext := fvm.NewScriptWithContextAndArgs(script, requestCtx, arguments...)
-	blockCtx := fvm.NewContextFromParent(
-		e.vmCtx,
-		fvm.WithBlockHeader(blockHeader),
-		fvm.WithDerivedBlockData(derivedBlockData))
-
-	err := func() (err error) {
-
-		start := time.Now()
-
-		defer func() {
-
-			prepareLog := func() *zerolog.Event {
-
-				args := make([]string, 0, len(arguments))
-				for _, a := range arguments {
-					args = append(args, hex.EncodeToString(a))
-				}
-				return e.logger.Error().
-					Hex("script_hex", script).
-					Str("args", strings.Join(args, ","))
+	var err error
+	defer func() {
+		prepareLog := func() *zerolog.Event {
+			args := make([]string, 0, len(arguments))
+			for _, a := range arguments {
+				args = append(args, hex.EncodeToString(a))
 			}
-
-			elapsed := time.Since(start)
-
-			if r := recover(); r != nil {
-				prepareLog().
-					Interface("recovered", r).
-					Msg("script execution caused runtime panic")
-
-				err = fmt.Errorf("cadence runtime error: %s", r)
-				return
-			}
-			if elapsed >= e.config.LogTimeThreshold {
-				prepareLog().
-					Dur("duration", elapsed).
-					Msg("script execution exceeded threshold")
-			}
-		}()
-
-		view := delta.NewDeltaView(snapshot)
-		return e.vm.Run(blockCtx, scriptInContext, view)
+			return e.logger.Error().
+				Hex("script_hex", script).
+				Str("args", strings.Join(args, ","))
+		}
+
+		elapsed := time.Since(startedAt)
+
+		if r := recover(); r != nil {
+			prepareLog().
+				Interface("recovered", r).
+				Msg("script execution caused runtime panic")
+
+			err = fmt.Errorf("cadence runtime error: %s", r)
+			return
+		}
+		if elapsed >= e.config.LogTimeThreshold {
+			prepareLog().
+				Dur("duration", elapsed).
+				Msg("script execution exceeded threshold")
+		}
```

Comment:
Not sure this part belongs in the defer.
It might be better o to check for timeout by looking at the error on requestCtx.

Reply:
looks like the timeout threshold (`e.config.ExecutionTimeLimit`) is different from the logging threshold (`e.config.LogTimeThreshold`).  I don't have the historical context here, but I presume the logging threshold is smaller the timeout threshold, in which case checking requestCtx is unnecessary.

I agree that combining the two would be a good idea